apps/recovery-portal/src/main/webapp/password-recovery.jsp

```diff
@@ -242,38 +233,21 @@
                 var errorMessage = $("#error-msg");
                 errorMessage.hide();
 
-                var isSaaSApp = JSON.parse("<%= isSaaSApp %>");
-                var tenantDomain = "<%= tenantDomain %>";
-                var isEmailUsernameEnabled = JSON.parse("<%= isEmailUsernameEnabled %>");
-
+                var tenantDomainInRequest = "<%= tenantDomain %>";
+                var tenantDomain = document.getElementById("tenantDomain");
+                if (tenantDomainInRequest) {
+                    tenantDomain.value = tenantDomainInRequest;
+                }
+                var isSaaSAppInRequest = "<%= isSaaSApp %>";
+                var isSaaSApp = document.getElementById("isSaaSApp");
+                if (isSaaSAppInRequest) {
+                    isSaaSApp.value = isSaaSAppInRequest;
+                }
```

Comment:
No need to re-assign these values from JS. Just assign the value to the hidden field directly.